app/javascript/packages/document-capture/components/in-person-location-full-address-entry-post-office-search-step.tsx

```diff
@@ -1,14 +1,136 @@
+  ) => {
+    setFoundAddress(address);
+    setLocationResults(locations);
+  };
+
+  // ref allows us to avoid a memory leak
+  const mountedRef = useRef(false);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  // useCallBack here prevents unnecessary rerenders due to changing function identity
+  const handleLocationSelect = useCallback(
+    async (e: any, id: number) => {
+      if (flowPath !== 'hybrid') {
+        e.preventDefault();
+      }
+      const selectedLocation = locationResults![id]!;
+      const { streetAddress, formattedCityStateZip } = selectedLocation;
+      const selectedLocationAddress = `${streetAddress}, ${formattedCityStateZip}`;
+      onChange({ selectedLocationAddress });
+      if (autoSubmit) {
+        setDisabledAddressSearch(true);
+        setTimeout(() => {
+          if (mountedRef.current) {
+            setDisabledAddressSearch(false);
+          }
+        }, 250);
+        return;
+      }
+      if (inProgress) {
+        return;
+      }
+      const selected = transformKeys(selectedLocation, snakeCase);
+      setInProgress(true);
+      try {
+        await request(LOCATIONS_URL, {
+          json: selected,
+          method: 'PUT',
+        });
+        // In try block set success of request. If the request is successful, fire remaining code?
+        if (mountedRef.current) {
+          setAutoSubmit(true);
+          setImmediate(() => {
+            e.target.disabled = false;
+            if (flowPath !== 'hybrid') {
+              trackEvent('IdV: location submitted', {
+                selected_location: selectedLocationAddress,
+              });
+              forceRedirect(inPersonURL!);
+            }
+            // allow process to be re-triggered in case submission did not work as expected
+            setAutoSubmit(false);
+          });
+        }
+      } catch {
+        setAutoSubmit(false);
+      } finally {
+        if (mountedRef.current) {
+          setInProgress(false);
+        }
+      }
+    },
+    [locationResults, inProgress],
+  );
 
   return (
     <>
+      {apiError && (
+        <Alert type="error" className="margin-bottom-4">
+          {t('idv.failure.exceptions.post_office_search_error')}
+        </Alert>
+      )}
       <PageHeading>{t('in_person_proofing.headings.po_search.location')}</PageHeading>
       <p>{t('in_person_proofing.body.location.po_search.po_search_about')}</p>
+      <FullAddressSearch
+        registerField={registerField}
+        onFoundLocations={onFoundLocations}
+        onLoadingLocations={setLoadingLocations}
+        onError={setApiError}
+        disabled={disabledAddressSearch}
+      />
+      {locationResults && foundAddress && !isLoadingLocations && (
+        <InPersonLocations
+          locations={locationResults}
+          onSelect={handleLocationSelect}
+          address={foundAddress?.address || ''}
```

Comment:
If we're checking `&& foundAddress` in the condition, does this need the `?.` operator?

Reply:
fixed in aad8a2b8ddbd7b88091507f516d2d01a212456b5